src/components/dashboard/index.js

```diff
@@ -0,0 +1,344 @@
+    } else {
+        desc = description;
+    }
+    return desc;
+};
+
+/**
+ * An item in the dashboard.
+ *
+ * @param {Object} props - The props for the component.
+ * @param {String} props.kind - The kind of item. Example: "app" or "analysis".
+ * @param {Object} props.content - The content for the item returned from the API.
+ * @returns {Object}
+ */
+export const DashboardItem = (props) => {
+    const classes = useStyles();
+    const { kind, content } = props;
+    const cardID = `${kind}-${content.id}`;
+
+    const description = cleanDescription(content.description);
+    const [origination, date] = getOrigination(kind, content);
+    const user = cleanUsername(content.username);
+
+    return (
+        <Card
+            className={classes.dashboardCard}
+            id={buildID(ids.ITEM_BASE, cardID)}
+            elevation={4}
+        >
+            <CardContent
+                classes={{
+                    root: classes.root,
+                }}
+            >
+                <Typography noWrap variant="h6" component="h6">
+                    {content.name}
+                </Typography>
+
+                <Typography
+                    classes={{ root: classes.subtitle }}
+                    gutterBottom
+                    noWrap
+                    color="textSecondary"
+                >
+                    {origination} {`${user} on ${date}`}
+                </Typography>
+
+                <Typography color="textSecondary" variant="body2" component="p">
+                    {description}
+                </Typography>
+            </CardContent>
+
+            <CardActions
+                classes={{
+                    root: classes.actionsRoot,
+                }}
+            >
+                <Button
+                    href={content.link}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    color="primary"
+                >
+                    {getMessage("open")}
+                </Button>
+            </CardActions>
+        </Card>
+    );
+};
+
+const DashboardSection = ({ name, kind, items }) => {
+    const classes = useStyles();
+
+    return (
+        <div className={classes.section}>
+            <Typography
+                noWrap
+                gutterBottom
+                variant="h5"
+                component="h5"
+                color="primary"
+            >
+                {name}
+            </Typography>
+
+            <Divider classes={{ root: classes.dividerRoot }} />
+
+            <div className={classes.sectionItems}>
+                {items.map((item) => (
+                    <DashboardItem kind={kind} content={item} key={item.id} />
+                ))}
+            </div>
+        </div>
+    );
+};
+
+const DashboardSkeleton = () => {
+    const classes = useStyles();
+    const numSkellies = [0, 1, 2];
+    const skellies = numSkellies.map((n) => (
+        <div className={classes.section}>
```

Comment:
One last thing since I just made a PR for this too - an ID here so the students can detect when loading begins and ends to ease testing.

Reply:
Okay, I wrapped the loading skeletons in a div and gave it a static ID. Let me know if I need to do more.

Reply:
Nope, that's perfect - thanks!